xarray/core/computation.py

```diff
@@ -1436,13 +1485,16 @@ def corr(da_a: T_DataArray, da_b: T_DataArray, dim: Dims = None) -> T_DataArray:
             "Only xr.DataArray is supported."
             f"Given {[type(arr) for arr in [da_a, da_b]]}."
         )
-
-    return _cov_corr(da_a, da_b, dim=dim, method="corr")
+    if weights is not None:
+        if not isinstance(weights, DataArray):
+            raise TypeError("Only xr.DataArray is supported." f"Given {type(weights)}.")
+    return _cov_corr(da_a, da_b, weights=weights, dim=dim, method="corr")
 
 
 def _cov_corr(
     da_a: T_DataArray,
     da_b: T_DataArray,
+    weights: T_DataArray = None,
```

Comment:
```suggestion
    weights: T_DataArray | None = None,
```

I think this is why `mypy` is failing

Reply:
Yeah, I pushed a change...

Reply:
(FYI for the suggested changes, GH lets you hit "Commit suggestion" to commit directly!)